### PR TITLE
Fix lint findings

### PR DESCRIPTION
**Pull request by andrewpollock**
This fixes the vast majority of the outstanding lint findings. 

I need to discuss/get some assistance with the outstanding ones.

```
************* Module importer
docker/importer/importer.py:213:4: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
************* Module osv.cache
osv/cache.py:87:14: W1505: Using deprecated method getcallargs() (deprecated-method)
osv/cache.py:70:0: C0103: Function name "Cached" doesn't conform to the `snake_case` group in the '^(?:(?P<exempt>setUp|tearDown|setUpModule|tearDownModule)|(?P<camel_case>_?[A-Z][a-zA-Z0-9]*)|(?P<snake_case>_?[a-z][a-z0-9_]*))$' pattern (invalid-name)

------------------------------------------------------------------
Your code has been rated at 9.99/10 (previous run: 9.99/10, +0.00)
```

### CODE REVIEW

**Comment by oliverchang** [comment]
Nice!! Can you also fix the lint action here: https://github.com/google/osv.dev/blob/master/.github/workflows/lint.yaml ? This just uses the referenced bash script, which may not be re-surfacing all the necessary errors. 

For `docker/importer/importer.py`, can you refactor some of the nested code into smaller helper functions? 

For 

>osv/cache.py:87:14: W1505: Using deprecated method getcallargs() (deprecated-method)

Are we able to use the recommended replacements on https://docs.python.org/3/library/inspect.html#inspect.getcallargs ? 

> osv/cache.py:70:0: C0103: Function name "Cached" doesn't conform to the `snake_case` group in the '^(?:(?P<exempt>setUp|tearDown|setUpModule|tearDownModule)|(?P<camel_case>_?[A-Z][a-zA-Z0-9]*)|(?P<snake_case>_?[a-z][a-z0-9_]*))$' pattern (invalid-name) 

Let's just rename this one. I think we just missed this through code review previously.

**Comment by andrewpollock** [comment]
> Nice!! Can you also fix the lint action here: https://github.com/google/osv.dev/blob/master/.github/workflows/lint.yaml ? This just uses the referenced bash script, which may not be re-surfacing all the necessary errors.

I think you're referring to what I've done in https://github.com/google/osv.dev/issues/521 so the answer is yes, in a separate PR. The fix is going to start causing hard fails on legacy lint errors, hence this PR.

> For docker/importer/importer.py, can you refactor some of the nested code into smaller helper functions?

Yep, that was slightly higher hanging fruit than the rest.

> For 
>
> osv/cache.py:87:14: W1505: Using deprecated method getcallargs() (deprecated-method)
>
> Are we able to use the recommended replacements on https://docs.python.org/3/library/inspect.html#inspect.getcallargs ?

I'll need a bit of help, this function is doing things beyond my immediate skill level.

> osv/cache.py:70:0: C0103: Function name "Cached" doesn't conform to the snake_case group in the '^(?:(?PsetUp|tearDown|setUpModule|tearDownModule)|(?P<camel_case>?[A-Z][a-zA-Z0-9]*)|(?P<snake_case>?[a-z][a-z0-9_]*))$' pattern (invalid-name)
>
> Let's just rename this one. I think we just missed this through code review previously.

Ack, done.

**Comment by another-rex** [comment]
> > For
> > osv/cache.py:87:14: W1505: Using deprecated method getcallargs() (deprecated-method)
> > Are we able to use the recommended replacements on https://docs.python.org/3/library/inspect.html#inspect.getcallargs ?
> 
> I'll need a bit of help, this function is doing things beyond my immediate skill level.

Looking at the example here: https://docs.python.org/3/library/inspect.html#inspect.BoundArguments
I think for this, you'll need to split it out into a few more lines, from:
```
      key = (*getcallargs(func, *args, **kwargs).items(), unique_f_key)
```
to
```
sig = inspect.signature(func)
# Passing through the arguments made to the function
bound_args = sig.bind(*args, **kwargs) 
# Making it hashable and combining it with the function name
key = (*bound_args.arguments.items(), unique_f_key) 
```

**Comment by andrewpollock** [comment]
Thanks so much for your assistance @another-rex 

This is now 100% lint-clean @oliverchang

File: docker/importer/importer.py [hunk 1]
```diff
@@ -210,8 +210,10 @@ def _process_updates_git(self, source_repo):
     changed_entries = set()
     deleted_entries = set()
 
-    if source_repo.last_synced_hash:
-      # Syncing from a previous commit.
+    def _sync_from_previous_commit(repo):
```
Comment (oliverchang) [inline docker/importer/importer.py:213]: Can we pull this out into a top level method? I **think** we just need to pass `source_repo` here as an extra arg?

**Review by oliverchang — APPROVED** [review]
thank you so much for cleaning this up! Just one last comment.

File: docker/importer/importer.py (hunk 1 above)
Comment (andrewpollock) [reply docker/importer/importer.py]: Done in bfc7965

**Review by oliverchang — APPROVED** [review]

**Pull request merged**